src/openforms/dmn/api/views.py

```diff
@@ -0,0 +1,172 @@
+from typing import List
+
+from django.utils.translation import gettext_lazy as _
+
+from drf_spectacular.utils import OpenApiParameter, extend_schema, extend_schema_view
+from rest_framework import authentication, permissions
+from rest_framework.exceptions import NotFound
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from openforms.api.drf_spectacular.functional import lazy_enum
+from openforms.api.mixins import ValidateQueryStringParametersMixin
+from openforms.api.serializers import ExceptionSerializer, ValidationErrorSerializer
+from openforms.utils.api.views import ListMixin
+
+from ..base import DecisionDefinition
+from ..registry import register
+from .serializers import (
+    DecisionDefinitionSerializer,
+    DecisionDefinitionVersionSerializer,
+    DecisionDefinitionXMLSerializer,
+    DecisionPluginSerializer,
+)
+
+
+def get_available_engines() -> List[str]:
+    return [engine.identifier for engine in register.iter_enabled_plugins()]
+
+
+ENGINE_PARAMETER = OpenApiParameter(
+    name="engine",
+    type=str,
+    location=OpenApiParameter.QUERY,
+    required=True,
+    description=_(
+        "Identifier of the decision engine to query. Note that some engines "
+        "may be disabled at runtime."
+    ),
+    enum=lazy_enum(get_available_engines),
+)
+
+DEFINITION_PARAMETER = OpenApiParameter(
+    name="definition",
+    type=str,
+    location=OpenApiParameter.QUERY,
+    required=True,
+    description=_("Identifier of the definition to retrieve available versions for."),
+)
```

Comment:
I noticed these 2 query params are required.

For the `/xml` endpoint you include the definiton as path-variable. Why do it as query param on other endpoints? Why is the engine not a path-variable? This feels inconsistent.

* `/api/v1/dmn/decision-definitions/versions` has required `engine` and `definition` **query** params.
* `/api/v1/dmn/decision-definitions/{definition}/xml` has required `engine` **query** param and (obviously) required `definition` **path** param
* `/api/v1/dmn/decision-definitions` has required `engine` **query** params - should it be required?